### PR TITLE
fix tests for agents-networks gis example

### PR DESCRIPTION
**Pull request by wang-boyu**
Addresses https://github.com/projectmesa/mesa-examples/issues/172#issuecomment-2324471001 as part of https://github.com/projectmesa/mesa-examples/issues/172.

### CODE REVIEW

**Review by EwoutH — APPROVED** [review]
Thanks! It seems like it’s uncovering a new issue with the AgentsAndNetworks model, but this one is fixed!

File: gis/agents_and_networks/src/model/model.py [hunk 1]
```diff
@@ -59,14 +59,14 @@ class AgentsAndNetworks(mesa.Model):
...
-        data_crs: str,
-        buildings_file: str,
-        walkway_file: str,
-        lakes_file: str,
-        rivers_file: str,
-        driveway_file: str,
-        num_commuters,
+        campus="ub",
+        data_crs="epsg:4326",
+        buildings_file="data/ub/UB_bld.zip",
```
Comment (EwoutH) [inline gis/agents_and_networks/src/model/model.py:64]: Actually, the new error might be related:

> E   pyogrio.errors.DataSourceError: '/vsizip/data/ub/UB_bld.zip' does not exist in the file system, and is not recognized as a supported dataset name.

**Comment by EwoutH** [comment]
It seems super().init needs to be called in the AgentsAndNetworks’s Agent

**Comment by wang-boyu** [comment]
For the Building agent: https://github.com/projectmesa/mesa-examples/blob/89f9379d40da872dbef68a4f4b608697845cf00d/gis/agents_and_networks/src/agent/building.py#L23

For model: https://github.com/projectmesa/mesa-examples/blob/89f9379d40da872dbef68a4f4b608697845cf00d/gis/agents_and_networks/src/model/model.py#L81

Am I missing something?

**Comment by EwoutH** [comment]
Something weird is happening here. 

> This error occurs because Python’s dictionary keys must be hashable, and instances of a class are not hashable by default unless you define a __hash__ method in the class.
> 
> In your code, you are trying to use an instance of Building as a key in a dictionary.

But let’s solve that in the next PR. This one already solves two issues!

**Review by EwoutH — APPROVED** [review]

**Comment by wang-boyu** [comment]
Probably because a `__eq__()` method is defined for Building but not a `__hash__()` method.

**Comment by wang-boyu** [comment]
Great! Another error to fix : )

**Comment by wang-boyu** [comment]
Should be working now

**Comment by EwoutH** [comment]
Awesome! Thanks for your persistence!

**Pull request merged**